File: 303-range-sum-query-immutable/303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.num_array = [nums[0]]
        for i in range(1, len(nums)):
            self.num_array.append(self.num_array[i-1] + nums[i])
        

    def sumRange(self, left: int, right: int) -> int:
        if not left:
            return self.num_array[right]
        
        
        return self.num_array[right] - self.num_array[left-1]
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 303-range-sum-query-immutable/303-range-sum-query-immutable.ts
class NumArray {
    public nums: number[]
    constructor(nums: number[]) {
        this.nums = nums
        for(let i = 1; i<this.nums.length; i++){
            this.nums[i] += this.nums[i-1]
        }
    }

    sumRange(left: number, right: number): number {
        return left ? this.nums[right] - this.nums[left-1] : this.nums[right]
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */